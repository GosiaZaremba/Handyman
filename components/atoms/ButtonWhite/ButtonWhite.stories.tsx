import React from 'react';
import {storiesOf} from '@storybook/react-native';
import {ButtonWhite} from './ButtonWhite';

storiesOf('atoms/ButtonWhite', module)
  .addDecorator(story => story())
  .add('default', () => (
    <ButtonWhite buttonText={'Button Text'} onPress={() => {}} />
  ));
