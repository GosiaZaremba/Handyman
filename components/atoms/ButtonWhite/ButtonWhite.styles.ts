import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  pressable: {
    borderRadius: 10,
    width: '100%',
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  pressed: {
    borderRadius: 10,
    width: '100%',
    height: 45,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F9F9F9',
  },
  buttonText: {
    color: Colors.font.green,
    fontFamily: 'OpenSans-Bold',
  },
});
