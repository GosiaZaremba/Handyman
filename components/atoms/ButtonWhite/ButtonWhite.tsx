import React from 'react';
import {Pressable, Text, View} from 'react-native';
import styles from './ButtonWhite.styles';

export type Props = {
  buttonText: string;
  onPress: () => void;
};

export const ButtonWhite: React.FC<Props> = ({buttonText, onPress}) => {
  return (
    <View style={styles.container}>
      <Pressable
        style={({pressed}) => (pressed ? styles.pressed : styles.pressable)}
        onPress={onPress}>
        <Text style={styles.buttonText}>{buttonText}</Text>
      </Pressable>
    </View>
  );
};
