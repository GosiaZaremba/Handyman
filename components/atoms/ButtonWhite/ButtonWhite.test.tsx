import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import {ButtonWhite} from './ButtonWhite';

const mockOnPress = jest.fn();

describe('Atom buttonWhite', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test('Component Snapshot', () => {
    const tree = renderer
      .create(
        <ButtonWhite
          onPress={mockOnPress}
          buttonText={'Button White'}
          testID="atom-buttonWhite"
          testOnly_pressed={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure text is passed correctly', () => {
    const {getByTestId} = render(
      <ButtonWhite
        onPress={mockOnPress}
        buttonText={'Button White'}
        testID="atom-buttonWhite"
        testOnly_pressed={false}
      />,
    );
    const addPhotoButton = getByTestId('atom-buttonWhite');
    expect(addPhotoButton).toHaveTextContent('Button White');
  });

  test('ensure onPress works correctly', () => {
    const {getByTestId} = render(
      <ButtonWhite
        onPress={mockOnPress}
        buttonText={'Button White'}
        testID="atom-buttonWhite"
        testOnly_pressed={false}
      />,
    );
    const buttonWhite = getByTestId('atom-buttonWhite');
    fireEvent.press(buttonWhite);
    expect(mockOnPress).toHaveBeenCalledTimes(1);
    expect(buttonWhite.props.style.opacity).toBe(undefined);
  });

  test('ensure pressable pressed works correctly', () => {
    const {getByTestId} = render(
      <ButtonWhite
        onPress={mockOnPress}
        buttonText={'Button White'}
        testID="atom-buttonWhite"
        testOnly_pressed={true}
      />,
    );
    const buttonWhite = getByTestId('atom-buttonWhite');
    fireEvent.press(buttonWhite);
    expect(buttonWhite.props.style.opacity).toBe(0.8);
  });
});
