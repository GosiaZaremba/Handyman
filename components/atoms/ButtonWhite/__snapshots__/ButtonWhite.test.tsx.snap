// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom buttonWhite Component Snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 8,
      "height": 45,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "height": 45,
        "justifyContent": "center",
        "opacity": 0.8,
        "width": "100%",
      }
    }
    testID="atom-buttonWhite"
  >
    <Text
      style={
        Object {
          "color": "#44b35e",
          "fontFamily": "OpenSans-Bold",
        }
      }
    >
      Button White
    </Text>
  </View>
</View>
`;
